pkg/util/marshal/query.go

```diff
@@ -310,16 +346,30 @@ func encodeStream(stream logproto.Stream, s *jsoniter.Stream) error {
 		s.WriteRaw(`"`)
 		s.WriteMore()
 		s.WriteStringWithHTMLEscaped(e.Line)
-		if len(e.StructuredMetadata) > 0 {
+
+		if categorizeLabels && (len(e.StructuredMetadata) > 0 || len(e.Parsed) > 0) {
```

Comment:
I think for consistency we should write empty object(`{}`) instead of skipping it when we have not structured metadata and parsed labels.

Reply:
Do you mean always serializing:
```json
{
    "structuredMetadata": {},
    "parsed": {}
}
```

If so, I think it's better not to do it. It will add a lot of overhead to the response.